packages/code-editor/src/CodeEditor/CodeEditor.tsx

```diff
@@ -36,35 +51,76 @@ const defaultCodeEditorOptions: EditorProps["options"] = {
   },
 };
 
+/** Languages we have custom validations for. */
+const languages = { xml: "xml" };
+const completionProviders: Record<
+  string,
+  (monaco: Monaco, elements?: Elements) => object
+> = {
+  xml: (
+    monaco: Monaco,
+    elements?: Elements, // needed for XML language
+  ) => getXmlCompletionProvider(monaco, elements),
+};
+const validationMarkers: Record<
+  string,
+  (
+    content: string,
+    editor: any,
+    monaco: Monaco,
+    schema?: string, // needed for XML language
+  ) => Promise<object[]>
+> = {
+  xml: (content: string, editor: any, monaco: Monaco, schema?: string) =>
+    getXmlValidationMarkers(content, editor, monaco, schema),
+};
+const keyDownListeners: Record<
+  string,
+  (event: any, editor: any, monaco: Monaco) => void
+> = {
+  xml: (event: any, editor: any, monaco: Monaco) =>
+    handleXmlKeyDown(event, editor, monaco),
+};
+const editorOptions: Record<string, object> = {
+  xml: xmlOptions,
+};
+
 /**
- * A wrapper to the React Monaco editor (https://github.com/suren-atoyan/monaco-react) with our styles.
+ * A wrapper to the <a href="https://github.com/suren-atoyan/monaco-react" target="_blank"><b>React Monaco editor</b></a> with our styles.
```

Comment:
Markdown is simpler & more cross-compatible (eg.  it'll work in IDEs)

![image](https://github.com/user-attachments/assets/af7945a2-3061-4127-838a-3699fca99549)

```suggestion
 * A wrapper to the [**React Monaco editor**](https://github.com/suren-atoyan/monaco-react) with our styles.
```

Reply:
I'll update this 🙏